dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -369,6 +369,68 @@ def diff(self, other_profile: NumericStatsMixin, options: Dict = None) -> Dict:
                 "and '{}'".format(cls.__name__, other_profile.__class__.__name__)
             )
 
+        self_bin_counts = self._stored_histogram["histogram"]["bin_counts"]
+        self_bin_edges = self._stored_histogram["histogram"]["bin_edges"]
+        regen_histogram = False
+        num_psi_bins = 10
+
+        if isinstance(self_bin_counts, np.ndarray) and isinstance(
+            self_bin_edges, np.ndarray
+        ):
+            regen_histogram = True
+
+        new_self_histogram = {"bin_counts": None, "bin_edges": None}
+        new_other_histogram = {"bin_counts": None, "bin_edges": None}
+
+        if regen_histogram:
+            len_self_bin_counts = 0
+            if len(self_bin_counts) > 0:
+                len_self_bin_counts = len(self_bin_counts)
+
+            # re-calculate `self` histogram
+            if not len_self_bin_counts == num_psi_bins:
+                histogram, hist_loss = self._regenerate_histogram(
+                    bin_counts=self._stored_histogram["histogram"]["bin_counts"],
+                    bin_edges=self._stored_histogram["histogram"]["bin_edges"],
+                    suggested_bin_count=num_psi_bins,
+                )
+                new_self_histogram["bin_counts"] = histogram["bin_counts"]
+                new_self_histogram["bin_edges"] = histogram["bin_edges"]
+
+            if len_self_bin_counts == num_psi_bins:
+                new_self_histogram["bin_counts"] = self._stored_histogram["histogram"][
+                    "bin_counts"
+                ]
+                new_self_histogram["bin_edges"] = self._stored_histogram["histogram"][
+                    "bin_edges"
+                ]
+
+            # re-calculate `other_profile` histogram
+            both_profiles_bin_edges_equal = (
+                other_profile._stored_histogram["histogram"]["bin_edges"]
+                == self._stored_histogram["histogram"]["bin_edges"]
+            ).all() == 0
```

Comment:
at the end of this you have `== 0` but the variable name suggests it is checking bin edges are all equal. I think you could either fix the variable name or the  `== 0` plus the usage below.

Reply:
this was how I resolved flake8 complaining... let me see what I can do to clean this up

Reply:
cleaned up a bit more